simd/src/Kokkos_SIMD_AVX2.hpp

```diff
@@ -73,6 +73,18 @@ class simd_mask<double, simd_abi::avx2_fixed_size<4>> {
   KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION explicit simd_mask(value_type value)
       : m_value(_mm256_castsi256_pd(_mm256_set1_epi64x(-std::int64_t(value)))) {
   }
+  template <class G,
+            std::enable_if_t<
+                std::is_invocable_r_v<value_type, G,
+                                      std::integral_constant<std::size_t, 0>>,
+                bool> = false>
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION constexpr explicit simd_mask(
+      G&& gen) noexcept
+      : m_value(_mm256_castsi256_pd(_mm256_setr_epi64x(
+            -std::int64_t(gen(std::integral_constant<std::size_t, 0>())),
```

Comment:
Why is it correct to negate the argument here?

Reply:
It's to set all bits in the lane to 1 when the mask is set to true (in this case, to produce `0xFFFFFFFFFFFFFFFFULL`). I guess this imposes an underlying assumption that setting the mask to be `true` is strictly equivalent to setting it to be `1` rather than any non-zero value.